test/utilities/miscellaneous.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import os from 'os';
+import {
+  sleep
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from '../environmentSettings';
+
+export async function pause(durationInSeconds: number): Promise<void> {
+  await sleep(durationInSeconds * EnvironmentSettings.getInstance().throttleFactor * 1000);
+}
+
+export function log(message: string) {
+  console.log(message);
+}
+
+export function currentOsUserName(): string {
```

Comment:
It's something I "borrowed" from SO.  See https://stackoverflow.com/questions/40424298/how-to-get-os-username-in-nodejs.

The `!` is the Non-null assertion operator (see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html).  I added them to placate TypeScript, but you make a good point.  I made a change so that only one `!` is used.